controllers/machinedeployment_rolling.go

```diff
@@ -162,7 +162,7 @@ func (r *MachineDeploymentReconciler) reconcileOldMachineSets(ctx context.Contex
 	// and cause timeout. See https://github.com/kubernetes/kubernetes/issues/16737
 	oldMSs, cleanupCount, err := r.cleanupUnhealthyReplicas(ctx, oldMSs, deployment, maxScaledDown)
 	if err != nil {
-		return nil
+		return err
```

Comment:
Would be good to get confirmation that this is what we want. In reading the code, I think we would actually want this error to bubble up, but someone with more expertise in this area should confirm.

Reply:
Hm, we should hold off onto this, given that I want to make sure we dig into the reasoning before we go into exponential backoff

Reply:
mm I suspect this might have sneaked in. by accident here https://github.com/kubernetes-sigs/cluster-api/pull/544/files#diff-6ce0da53e972cfd7de6bb7f7ea3675fda5ec56348c5112b82c76566938aa759bR149